saleor/payment/utils.py

```diff
@@ -25,27 +32,161 @@
+            checkout_info,
+            lines,
+            line_info,
+            address,
+            discounts,
+        )
+        unit_gross = unit_price.undiscounted_price.gross.amount
+
+        quantity = line_info.line.quantity
+        product_name = f"{line_info.variant.product.name}, {line_info.variant.name}"
+        product_sku = line_info.variant.sku
+        line_items.append(
+            PaymentLineData(
+                quantity=quantity,
+                product_name=product_name,
+                product_sku=product_sku,
+                variant_id=line_info.variant.id,
+                gross=unit_gross,
+            )
+        )
+    shipping_amount = manager.calculate_checkout_shipping(
+        checkout_info=checkout_info,
+        lines=lines,
+        address=address,
+        discounts=discounts,
+    ).gross.amount
+
+    line_items.append(create_shipping_payment_line_data(amount=shipping_amount))
+
+    voucher_line_item = create_checkout_voucher_payment_line_data(checkout)
+    if voucher_line_item:
+        line_items.append(voucher_line_item)
+
+    return line_items
+
+
+def create_order_payment_lines_information(order: Order) -> List[PaymentLineData]:
+    line_items = []
+    for order_line in order.lines.all():
+        product_name = f"{order_line.product_name}, {order_line.variant_name}"
+
+        variant_id = order_line.variant_id
+        if variant_id is None:
+            continue
+
+        line_items.append(
+            PaymentLineData(
+                quantity=order_line.quantity,
+                product_name=product_name,
+                product_sku=order_line.product_sku,
+                variant_id=variant_id,
+                gross=order_line.unit_price_gross_amount,
+            )
+        )
+
+    line_items.append(
+        create_shipping_payment_line_data(amount=order.shipping_price_gross_amount)
+    )
+
+    voucher_line_item = create_order_voucher_payment_line_data(order)
+    if voucher_line_item:
+        line_items.append(voucher_line_item)
+
+    return line_items
+
+
+# Values are outside of model's pk range to resolve
+# any collision with actual product variant pk
+VOUCHER_PAYMENT_LINE_ID = 0
+SHIPPING_PAYMENT_LINE_ID = -1
+
+
+def create_shipping_payment_line_data(amount: Decimal) -> PaymentLineData:
+    return PaymentLineData(
+        quantity=1,
+        product_name="Shipping",
+        product_sku="Shipping",
+        variant_id=SHIPPING_PAYMENT_LINE_ID,
+        gross=amount,
+    )
+
+
+def create_checkout_voucher_payment_line_data(
+    checkout: Checkout,
+) -> Optional[PaymentLineData]:
+    discount_amount = -checkout.discount_amount
+    return create_voucher_payment_line_data(discount_amount)
+
+
+def create_order_voucher_payment_line_data(
+    order: Order,
+) -> Optional[PaymentLineData]:
+    discount_amount = order.total_gross_amount - order.undiscounted_total_gross_amount
+    return create_voucher_payment_line_data(discount_amount)
```

Comment:
Why do we need this? I am not sure that we should place them as a payment line. Personally I would say that this is something related to NP.